gobblin-metastore/src/main/java/gobblin/metastore/DatabaseJobHistoryStore.java

```diff
@@ -134,51 +135,53 @@
+      "SELECT * FROM gobblin_job_executions WHERE job_id IN (%s)";
 
   private static final String TASK_EXECUTION_QUERY_STATEMENT_TEMPLATE =
-      "SELECT * FROM gobblin_task_executions WHERE job_id=?";
-
-  private static final String JOB_METRIC_EXIST_QUERY_STATEMENT_TEMPLATE =
-      "SELECT * FROM gobblin_job_metrics " + "WHERE job_id=? AND metric_group=? AND metric_name=? AND metric_type=?";
-
-  private static final String TASK_METRIC_EXIST_QUERY_STATEMENT_TEMPLATE =
-      "SELECT * FROM gobblin_task_metrics " + "WHERE task_id=? AND metric_group=? AND metric_name=? AND metric_type=?";
+      "SELECT * FROM gobblin_task_executions WHERE job_id IN (%s)";
 
   private static final String JOB_METRIC_QUERY_STATEMENT_TEMPLATE =
-      "SELECT metric_group,metric_name,metric_type,metric_value FROM gobblin_job_metrics WHERE job_id=?";
+      "SELECT job_id,metric_group,metric_name,metric_type,metric_value FROM gobblin_job_metrics WHERE job_id IN (%s)";
 
   private static final String TASK_METRIC_QUERY_STATEMENT_TEMPLATE =
-      "SELECT metric_group,metric_name,metric_type,metric_value FROM gobblin_task_metrics WHERE task_id=?";
-
-  private static final String JOB_PROPERTY_EXIST_QUERY_STATEMENT_TEMPLATE =
-      "SELECT * FROM gobblin_job_properties WHERE job_id=? AND property_key=?";
-
-  private static final String TASK_PROPERTY_EXIST_QUERY_STATEMENT_TEMPLATE =
-      "SELECT * FROM gobblin_task_properties WHERE task_id=? AND property_key=?";
+      "SELECT job_id,m.task_id,metric_group,metric_name,metric_type,metric_value FROM gobblin_task_metrics m JOIN gobblin_task_executions t ON t.task_id = m.task_id WHERE job_id IN (%s)";
 
   private static final String JOB_PROPERTY_QUERY_STATEMENT_TEMPLATE =
-      "SELECT property_key, property_value FROM gobblin_job_properties WHERE job_id=?";
+      "SELECT job_id,property_key,property_value FROM gobblin_job_properties WHERE job_id IN (%s)";
 
   private static final String TASK_PROPERTY_QUERY_STATEMENT_TEMPLATE =
-      "SELECT property_key, property_value FROM gobblin_task_properties WHERE task_id=?";
-
-  private static final Timestamp DEFAULT_TIMESTAMP = new Timestamp(1000L);
-  private static final int DEFAULT_VALIDATION_TIMEOUT = 5;
+      "SELECT job_id,p.task_id,property_key,property_value FROM gobblin_task_properties p JOIN gobblin_task_executions t ON t.task_id = p.task_id WHERE job_id IN (%s)";
 
   private final DataSource dataSource;
 
+  private boolean isDatabaseUpToDate;
+
   @Inject
   public DatabaseJobHistoryStore(DataSource dataSource) {
     this.dataSource = dataSource;
+    try {
+      ImmutablePair<MigrationVersion, MigrationVersion> databaseVersions = getDatabaseVersions(dataSource);
+      this.isDatabaseUpToDate = databaseVersions.getValue().compareTo(databaseVersions.getKey()) >= 0;
+      if (!this.isDatabaseUpToDate) {
+        LOGGER.warn(String.format("%s is disabled because the database version %s is not at least %s. " +
```

Comment:
since users have to explicitly turn on the `JobHistoryStore` its probably better to just fail the entire job until they have done the migrations, otherwise they may not realize something is amiss until they see there past `x` executions haven't been recorded